### ISSUE TITLE
Adding of ObjectResolver and ObjectCollectionResolver for action parameters

### ISSUE THREAD

**Issue by jaimesangcap** [symfony/symfony#19381]
with the release of the ArgumentResolver

http://symfony.com/doc/current/cookbook/controller/argument_value_resolver.html

I wonder if its possible now to have built-in resolver for

array to object
nested array to nested object (maybe by annotations)
array to collection of object  (maybe by annotations)

for example I have 

```
class LogInUserCommand
{
    private $username;
    private $password;
    private $rememberMe;
}
```

then it will be resolved automatically when I typehint it in the controller action

```
public function loginAction(LogInUserCommand $command)
```

**Comment by iltar** [comment]
In theory you can make this, yes.

**Comment by Daskul** [comment]
yeah Im thinking of annotation mapping, and hopefully it can also be integrated with the validation after resolving the entity and return a response. not really sure how difficult it will be though.

**Comment by kix** [comment]
First, such a feature should definitely require an annotation, because if such a behaviour is introduced automagically (like, as a framework core feature), it gets quite hard to figure out why is the object hydration happening, for an unprepared user. Since adding complexity to the framework feels like a no-go, then annotation it is. What could it look like? How would you make it fit all the usecases? Such an annotation would be quite cumbersome, as it currently appears to me.

Second, this feature also requires having a way to figure out the request format and deserialize the request here. Looks like what you've proposed here can mostly be applied to query parameters, however we should think of JSON or maybe XML requests or whatnot. And since handling the request body and formulating the exchange format is totally up to the end user, or it could even be multi-format, it doesn't look like there's some common way of figuring this out. Imagine what a built-in resolver would have to deal with, since a core feature needs to support all the variety of possible usages. I'd say there's way too many of those here.

Third, I don't think I would even implement this myself in my userland code, as the harm from such magic would be much greater than the profit. Having to support different formats would make me juggle the `onKernelRequest` listener priorities, so that the request first gets deserialized, and only then it hits the hydrator. Also, I would have to somehow either disable or reprioritize the built-in Doctrine entity listener, making sure nothing gets in the way of my own hydration listener.

To sum this up, I think that this proposal shouldn't be a core feature of Symfony. The `Request` object already has all the attributes you need, and you have the form component to process and validate user requests.

**Comment by fabpot** [comment]
Closing as it probably does not belong to core.
